Mo & Joel - Targets workshop (#600)
ropensci/roweb3: content/blog/2023-07-06-teaching-targets-with-penguins/index.en.md

Reviewer: I think that for the English post (and for now), the file is just `index.md` (cf Blog Guide section on [Multilingual posts](https://blogguide.ropensci.org/authortechnical.html#multilingual-posts)) @maelle, is that correct?

Reviewer 2: Yes, the name for the English is index.md :-)

---
title: Teaching Targets with Penguins
author:
  - Athanasia Mo Mowinckel
  - Joel Nitta
date: '2023-07-06'
slug: teaching-targets-with-penguins
categories:
  - blog
tags:
  - targets
  - teaching
  - meetings
  - community

> ---

Author: @steffilazerte is it ok to add the preface key later, as I think the cross-post will happen after rOpenSci post, if I understand carpentry procedure correct.

> ---

Author: preface added after confirmation from Carpentries.

However, even though bringing up a different example may seem trivial to someone who is familiar with targets, we realized it adds considerable cognitive load to ask a learner to do such context switching.
So we are currently revising the lesson to stick to the penguin theme as much as possible, while still providing useful code examples.

## First workshop

### Setup

In the days before the workshop, we (Mo and Joel) communicated and revised the lesson plan together through a series of GitHub Pull requests.
As Mo had no experience with targets, but lots of R and teaching R experience, she could provide some feedback on the general flow of the lesson.
Joel, with more experience in using and teaching targets, could provide context and intention of the lesson episodes.
This made a really nice dynamic between the two of us, and made the total experience even more enjoyable.

The workshop was hosted by the [Digital Scholarship Center](https://www.ub.uio.no/english/libraries/dsc/) (DSC) at the University of Oslo, which also hosts the local [Carpentries community (Carpentry@UiO)](https://www.ub.uio.no/english/libraries/dsc/carpentry-uio/).

> The DSC took care of the important administrative tasks when hosting a workshop; setting up a website to announce the workshop, a sign-up solution (with waiting list), booking an accessible and suitable room, etc.

Reviewer: ```suggestion
The DSC took care of the important administrative tasks required to host a workshop: setting up a website to announce the workshop, handling the sign-ups (including a waiting list), booking an accessible and suitable room, etc.
```

> They additionally provided sticky-notes and extension cords for the room on the day.

Reviewer: ```suggestion
Additionally, they provided sticky-notes and extension cords for the room on the day.
```

> In short, organising the workshop was made very smooth by the expert and friendly staff of the DSC.

Reviewer: Wish I had a DSC for when I give workshops! 😆

Author: right? They are awesome, they make the whole process so much easier for us instructors.

### Participants

> We had 19 participants sign up, and a little over half came (about normal no-show rate for DSC workshops). The participants covered a wide range of disciplines, from psychology to biology and humanities, and in every academic stage.  This varied group was perfect to test the workshop on!

Reviewer: ```suggestion
We had 19 participants sign up, and more than half of those who signed up came (about normal no-show rate for DSC workshops). The participants covered a wide range of disciplines, from psychology to biology to humanities, and in every academic stage.  This diverse group was perfect to test the workshop on!
```

### Feedback

Overall, feedback was quite positive, so we feel that we are definitely on the right track with this implementation of the workshop. Participants said that they feel more confident about using targets for their own analyses. The biggest challenge seems to be writing functions, which may be unfamiliar to many researchers but is absolutely necessary for effective use of targets. This is something that probably needs to be tailored to each workshop: the instructor should understand the participants' skill level with writing functions and provide more instruction on this topic as needed.

## Next steps
We are very happy with the success of the first (unofficial) [Carpentries targets workshop](https://github.com/joelnitta/targets-workshop).
Our biggest hope is that more people will teach the workshop and contribute back to it.
If you have any suggestions about the material, please feel free to submit a PR or [file an issue](https://github.com/joelnitta/targets-workshop/issues), or just reach out to us by email.

Happy workflowing!

> ![Joel and Mo after the workshop](mo-joel.jgp)

Reviewer: Use Hugo codes and alt text (cf [Blog Guide images](https://blogguide.ropensci.org/authortechnical.html#images---general))
```suggestion
{{< figure src = "mo-joel.jgp" alt = "Joel and Mo posing with grins after the workshop in front of a large projector screen showing the rOpenSci website behind them." caption = "Joel and Mo after the workshop" class = "center" >}}
```
